frontend/src/pages/modelServing/screens/metrics/MetricsChart.tsx

```diff
@@ -0,0 +1,128 @@
+import { ModelServingMetricsContext } from './ModelServingMetricsContext';
+import { CubesIcon } from '@patternfly/react-icons';
+import { ContextResourceData, PrometheusQueryRangeResultValue } from '../../../../types';
+import { TimeframeTime } from '../const';
+
+type MetricsChartProps = {
+  title: string;
+  color: string;
+  metrics: ContextResourceData<PrometheusQueryRangeResultValue>;
+  unit?: string;
+  threshold?: number;
+};
+
+const MetricsChart: React.FC<MetricsChartProps> = ({ title, color, metrics, unit, threshold }) => {
+  const bodyRef = React.useRef<HTMLDivElement>(null);
+  const [chartWidth, setChartWidth] = React.useState(0);
+  const { currentTimeframe, lastUpdateTime } = React.useContext(ModelServingMetricsContext);
+
+  const processedData = React.useMemo(
+    () =>
+      metrics?.data.map((data) => ({
+        x: data[0] * 1000,
+        y: parseInt(data[1]),
+        name: title,
+      })),
+    [metrics, title],
+  );
+
+  const maxValue = Math.max.apply(
+    null,
+    processedData.map((e) => e.y),
+  );
+
+  const hasData = processedData.length > 0;
+
+  React.useEffect(() => {
+    const ref = bodyRef.current;
+    let observer: () => void;
+    if (ref) {
+      const handleResize = () => {
+        setChartWidth(ref.clientWidth);
+      };
+      observer = getResizeObserver(ref, handleResize);
+      handleResize();
+    }
+    return () => observer();
+  }, [bodyRef]);
+
+  return (
+    <Card>
+      <CardTitle>{`${title}${unit ? ` (${unit})` : ''}`}</CardTitle>
+      <CardBody style={{ height: hasData ? 400 : 200, padding: 0 }}>
+        <div ref={bodyRef}>
+          {hasData ? (
+            <Chart
+              ariaTitle={title}
+              containerComponent={
+                <ChartVoronoiContainer
+                  labels={({ datum }) => `${datum.name}: ${datum.y}`}
+                  constrainToVisibleArea
+                />
+              }
+              domain={{ y: maxValue === 0 ? [-1, 1] : [0, maxValue + 1] }}
+              height={400}
+              width={chartWidth}
+              padding={{ left: 70, right: 50, bottom: 70, top: 50 }}
+              themeColor={color}
+            >
+              <ChartAxis
+                tickCount={10}
+                tickFormat={(x) => convertTimestamp(x, formatToShow(currentTimeframe))}
+                domain={{
+                  x: [lastUpdateTime - TimeframeTime[currentTimeframe] * 1000, lastUpdateTime],
+                }}
+                fixLabelOverlap
+              />
+              <ChartAxis dependentAxis tickCount={10} fixLabelOverlap />
+              <ChartGroup>
+                <ChartArea data={processedData} />
+              </ChartGroup>
+              {threshold && <ChartThreshold data={getThresholdData(processedData, threshold)} />}
+            </Chart>
+          ) : (
+            <EmptyState>
+              {metrics.loaded ? (
+                <>
+                  <EmptyStateIcon icon={CubesIcon} />
```

Comment:
PF doesn't have obvious guidance on this one. There is this option, but I don't think its quite appropriate here:
<img width="694" alt="Screenshot 2023-02-10 at 9 49 59 AM" src="https://user-images.githubusercontent.com/53235442/218121778-08c7276c-5485-4c41-bd81-0335089b3200.png">

@vconzola Do you have any thoughts on this?

Reply:
Personally I would prefer not to use the magnifying glass

Reply:
I guess my question would be, what would cause us to get "no data ever for a chart"? If it's some sort of backend error, we should use the error empty state. Knowing what's causing the state will help determine what the UI should present.

Reply:
Another question for @DaoDaoNoCode or @lucferbux... is there anything a user can do to make information in the chart visible? Is it because of a configuration they did ahead of time that the information isn't available, or do they just need to wait for information to be collected to display it?

Reply:
As I understand it -- the data is just flat out not in prometheus -- not out of range, just never updated. So you had not yet hit the endpoint for instance.

Reply:
I'm almost positive there's a "no data available" pattern floating around in console dot UIs. Lemme ask some other designers.

Reply:
We'll address this via another PR when (if?) you guys come to a desired change.